### PR TITLE
✨ Source File: support ZIP file

### PR DESCRIPTION
**Pull request by mrhydejc**
## What
If target file ends with .zip, unzip it, before parsing.
Issue : https://github.com/airbytehq/airbyte/issues/32352

## How
Consider file as binary (using tmp cache) then use Python's zipfile to extract-all in a temp dir. Process only first file.

### CODE REVIEW

File: airbyte-integrations/connectors/source-file/source_file/client.py
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-file/source_file/client.py:455]: Any reason for this change?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Thanks for the contribution @mrhydejc and sorry for the delay to start the review process.

File: airbyte-integrations/connectors/source-file/source_file/client.py
Comment (mrhydejc) [reply airbyte-integrations/connectors/source-file/source_file/client.py:455]: zipfile.ZipFile need a file path as string. TemporaryFile does not provide a path.

**Comment by marcosmarxm** [comment]
@mrhydejc thanks, I requested to the connector team make the final review of the contribution.

**Comment by lazebnyi** [comment]
Hey @mrhydejc 
Can you allow me to edit PR files. I need this to do all test before merge.

**Comment by mrhydejc** [comment]
Hi @lazebnyi,
"Allow edits and access to secrets by maintainers" is checked and I invited you as collaborator in my repo. Is it enough ?

**Review by lazebnyi — APPROVED** [review]

**Review by marcosmarxm — APPROVED** [review]
LGTM

**Pull request merged**
